library/alloc/src/vec/is_zero.rs

```diff
@@ -17,12 +17,16 @@ macro_rules! impl_is_zero {
     };
 }
 
+/************ Primitives ************/
+
+impl_is_zero!(i8, |x| x == 0);
 impl_is_zero!(i16, |x| x == 0);
 impl_is_zero!(i32, |x| x == 0);
 impl_is_zero!(i64, |x| x == 0);
 impl_is_zero!(i128, |x| x == 0);
 impl_is_zero!(isize, |x| x == 0);
 
+impl_is_zero!(u8, |x| x == 0);
```

Comment:
We can add comments -- but u8/i8 are already done by https://github.com/rust-lang/rust/blob/master/library/alloc/src/vec/spec_from_elem.rs#L20-L48, I think.

Reply:
1. Is there any reason why we want to specialize for i8/u8 instead of specializing on Copy types and probably checking for size_of::<T>() == 1? Compiler manages replace iteration by memset. https://godbolt.org/z/rzYYYhTKj There is little difference in generated code but it is still almost similar.
2. This implementation in this file allow [u8; N] and [i8;N] to be IsZero too and need less repetition than implementing IsZero for byte arrays directly.

Reply:
Just because it's size-1 & Copy doesn't mean it's legal to branch on the value, though.

Reply:
@scottmcm I mean something like this:

```
impl<T: Copy>impl SpecFromElem for T {
    fn from_elem(elem: T, n:usize)->Vec<T>{
        if core::mem::size_of::<T>() == 1 {
              let mut v = Vec::with_capacity(n);
              unsafe{
                     let byte_val: u8 = ptr::read(&elem as const T* as const u8*);
                     ptr::write_bytes(v.as_mut_ptr(), byte_val, n);
                     v.set_len(n);
              }
              return v;
        }
        // Default impl
        ...
    }
}
```

And probably IsZero variant for u8/i8.

Reply:
It's not generally valid to read a 1-byte value as u8 (I guess *maybe* with `MaybeUninit<u8>` that could be OK?) But I'm not sure this is worth trying to optimize for ourselves; I'd hope that LLVM can lower our standard init loop. (Or can be convinced to do so). I think slice::fill for example does pretty OK without such shenanigans.

Reply:
Valid, didn't think about uninit values before.